docs/_docs/sql-reference/explain-help-with-performance-problems.adoc

```diff
@@ -0,0 +1,495 @@
+Extra **Sort** operator brings additional performance costs, we can avoid it also
+with help from **HINTS** side:
+[source,sql]
+----
+SELECT /*+ FORCE_INDEX(t1_col1_idx, t2_col1_idx) */ t1.id as t1id, t2.id as t2id FROM t1 JOIN t2 USING (col1);
+----
+And possible plan becomes:
+[source,sql]
+----
+Project
+  MergeJoin
+    Exchange
+      IndexScan
+          index: T1_COL1_IDX
+    Exchange
+      IndexScan
+          index: T2_COL1_IDX
+----
+== Performance impact of correlated subqueries
+SQL-99 allows for nested subqueries at nearly all places within a query.
+Ignite 3 supports nested subqueries, it can be correlated or not. Performance of certain complex correlated subqueries may be insufficient. Let`s consider a correlated query:
+[source,sql]
+----
+CREATE TABLE emp(dept_id INTEGER PRIMARY KEY, name VARCHAR, salary INTEGER);
+CREATE TABLE dept(id INTEGER PRIMARY KEY, name VARCHAR);
+
+SELECT emp.name, (SELECT dept.name FROM dept WHERE emp.dept_id=dept.id)
+FROM emp
+WHERE emp.salary > 1000;
+----
+We can see nested correlated subquery here, lets check the plan:
+[source,sql]
+----
+Project
+  CorrelatedNestedLoopJoin
+    Exchange
+      TableScan
+          table: PUBLIC.EMP
+          filters: >(SALARY, 1000)
+    ColocatedHashAggregate
+      Exchange
+        TableScan
+            table: PUBLIC.DEPT
+            filters: =($cor1.DEPT_ID, ID)
+----
+In Ignite 3 there is present some performance issues related to correlated subqueries and as a result:
+
+- Such subqueries may become bottlenecks.
+- Queries over even small tables can cause high CPU and memory consumption.
+- Certain queries may perform slower than expected.
+
+Plan shows that **CorrelatedNestedLoopJoin** operator is present. Thus if some performance issues are found in such a case, it would be more
+efficient to rewrite for equal query but without nested subquery, like:
+[source,sql]
+----
+SELECT emp.name, dept.name
+FROM emp, dept
+WHERE emp.salary > 1000 AND emp.dept_id=dept.id;
+----
+And new plan becomes:
+[source,sql]
+----
+Exchange
+  Project
+    HashJoin
+        condition: =(DEPT_ID, ID)
+      TableScan
+          table: PUBLIC.EMP
+          filters: >(SALARY, 1000)
+      TableScan
+          table: PUBLIC.DEPT
+----
+Which performs match better than previous one.
+
+== Excessive sort
+Lets explain we have an index involved two columns one of them is participate in predicate and other in ordering, or in sql terms:
+[source,sql]
+----
+CREATE TABLE emp(dept_id INTEGER PRIMARY KEY, name VARCHAR, salary INTEGER);
+CREATE INDEX emp_salary_name_idx ON emp(salary, name);
+SELECT dept_id FROM emp WHERE salary = 1 ORDER BY name;
+----
+Expectations:
+
+- Index need to be used here.
+- No additional sort is needed because index is ordered by **name** column is satisfies initial query ordering.
+
+But plan can shows:
+[source,sql]
+----
+Project
+  Exchange
+    Sort
```

Comment:
It is not clear why Sort is under Exchange?
Should collation attribute for Exchange node be here? to emphasize the Exchange is aware of sorted input and preserve the collation?